### PR TITLE
Fix OS detection in a docker container

### PR DESCRIPTION
**Pull request by EchoPouet**
Hello,

I create this request to fix the OS detection in the docker containers. **distros** module use the command **uname -a** to get the OS name but in a docker container the value returned is not the container OS but the host OS.

So it's why I changed the code to use the command **cat /etc/os-release | grep ^ID=** to get the OS ID (see https://www.cyberciti.biz/faq/how-to-check-os-version-in-linux-command-line/). The OS ID is normally unique. An example with a CentOS container run in Ubuntu (the host):
```
root@bubuntu: docker run -t -i --rm centos bash
[root@95af48f1547c /]# uname -a
Linux 95af48f1547c 5.3.0-24-generic #26~18.04.2-Ubuntu SMP Tue Nov 26 12:34:22 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
[root@95af48f1547c /]# cat /etc/os-release | grep ^ID=
ID="centos"
```
I tested this fix in following OS:
- Elementary,
- Ubuntu,
- Debian,
- Fedora,
- OpenMandriva,
- CentOS
- Alpine,
- Mageia,
- Zorin,
- RedHat,
- ArchLinux.

Also, I created a test to check if the OS detection is good. It's very simple but it exists.

Thanks

### CODE REVIEW

File: lib/pure/distros.nim [hunk 1]
```diff
@@ -154,32 +160,37 @@ proc detectOsImpl(d: Distribution): bool =
...
-  of Distribution.Solaris:
-    let uname = toLowerAscii(uname())
-    result = ("sun" in uname) or ("solaris" in uname)
-  of Distribution.Haiku:
-    result = defined(haiku)
   else:
-    let dd = toLowerAscii($d)
-    result = dd in toLowerAscii(uname()) or dd in toLowerAscii(release()) or
-              ("operating system: " & dd) in toLowerAscii(hostnamectl())
+    when defined(linux):
```
Comment (Araq) [inline lib/pure/distros.nim]: Huh? That cannot be correct for Haiku and the BSDs...

**Comment by Araq** [comment]
Really nice job, but please somehow ensure you only changed the logic for the OSes that you have tested it for.

**Comment by EchoPouet** [comment]
Thank you but I don't know why I change the detection for Haiku and BSD. Maybe a merge error :)

**Comment by EchoPouet** [comment]
Can I expected this fix merged in the next release ? Because I work every-time in docker (CI/CD use docker also) and I would like remove this fix in my code.

Thanks

**Comment by Araq** [comment]
I would love to merge it but please ensure the BSDs keep working.

**Comment by EchoPouet** [comment]
Ho I missed your request, I'm going to test my change in BSD.

**Pull request merged**